### PR TITLE
Adds Netdem notebooks to repository

### PR DESCRIPTION
**Pull request by samgdotson**
This PR adds net demand prediction notebooks and updates param_string function to include prediction window size.

### CODE REVIEW

File: twoinputs/netdemand-solarangle-hourly.ipynb [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
...d9QjiuE5Ph9gwDKQRx/M80obg4TGV2bwkO7UX9eYSSnHz2nbuBIN+0/wR8LeFEO9tuEwKIY4A/wvwhUHuJEikflEI8boQ4qQQ4jdarv+nQohCw79NIcQXhRDnhBAvCCH2N1z3vweXvyOE+PGGy38iuOycEOLXB3x+EXeBI7tSnJnPk6vYSAmZ2DbpITb9H4CobDoi4t7i1FyOcs3l8f3Dd3tXmji6O43rSd65ke+/8Tp5bXaF/WMJhhPt38+e5yGEWL8grlaRllkv7duJJ6TbnfBkOhTE3tAQ9syMnzYdnGRPBr2urWXT+Zr/73RQMi2lJJ/Pc+3atXpPYjdEkGTraRscqhULHeLewVpN+yIEIthfUbP7CuLWz0GlUmkSxNVqtS6Iw+qKUCTLpRUKVpJkcrB6ac/zMAyD8fFxHnroIY4fnKLsKixmC00BXgDfu+ry8rzLm0sef6nNkLclc6+d4ysXHT7/ju2PzqrVsN58k+qRI8hkEi+VonLsIay33kTJr32vCCHqoVrFmkvC7CGIYxubt7JatpsCtaSUyIqfWC8No0m8NaJpGpqmdQ3W2okOcZjevx5BHNMEjue3PIQ9yK2LV1oig5QeKR0eHFVQBLyW2AOAObu95xEPehT8n8DbwHPA2eCyPwLeDP799wa8nyrwISnlceAR4CeEEE8BCCEeA1qXxv46sCKlPAT8Q+DvB9seBT4NPAj8BPDPhRCqEEIF/h/gJ4GjwC8F20ZsQY5MJilUnHrv23ZziAtR0nRExD1FWC59dPfWKJcOOTSRRFeVDe8jXi3ZXFgsdiyXBuolpI1BQoMgKhWk6ZcxdptFG3F3CcWoGZS2A1SPPoi6fBNtfh4AXRWMxgTzpWahkQ80ZyI4l9Z1nevXr1OtVruKlhARlB1vtEOcjukIsf6SacUKBLHTXxA3OsJhsnEngWwHY5waF5O8m8tkExmGjME+C47jEIvF6vt5ZGqYZDJJevoguq5TCYTi2RWP08seT+xS+W+PG/zakxmGDuzmI5XLfHCPiuP5vcfWqdOIapXyiRP1xyg9+QR4ktjLL9cva5ylnK/YdYOgE37JtL0uUdYL3yFuHrkkA/HvBKK3G5Zlde0j3qmCOB6Pr2skXiL4aBYdv3rBMNoXQd1gLFNCh7guODKs8IIYwzVMzG0erDXoHOIy8DTwq8DzwDeBl4C/AXxUSjn
```
Comment (LukeSeifert) [inline twoinputs/netdemand-solarangle-hourly.ipynb]: The spacing between these numbers is a bit inconsitent.

Comment (LukeSeifert) [inline twoinputs/netdemand-solarangle-hourly.ipynb]: Is this intended? It seems like after this point the outputs aren't working.

Comment (LukeSeifert) [inline twoinputs/netdemand-solarangle-hourly.ipynb]: Outputs work again after this point.

Comment (LukeSeifert) [inline twoinputs/netdemand-solarangle-hourly.ipynb]: Extra # here?

Comment (LukeSeifert) [inline twoinputs/netdemand-solarangle-hourly.ipynb]: Here as well an extra #? Maybe intentional

**Review by LukeSeifert — COMMENTED** [review]
Looks good, I made a few comments but nothing too serious. There is one spot where it looks like you had a keyboard interrupt, not sure about that part.

File: net_demand.ipynb [hunk 2]
```diff
@@ -0,0 +1,1124 @@
...
+    "\n",
+    "The following items are included:\n",
+    "1. An initial demonstration of the ESN with a 72-hour prediction.\n",
+    "2. Grid optimization over spectral radius and noise\n",
+    "3. Grid optimization over reservoir size and sparsity\n",
+    "4. Demonstrate the effect of training length on accuracy.\n",
+    "5. A final demonstration of the ESN with optimized hyper-parameters.\n",
+    "\n",
+    "\n",
+    "The net demand prediction might be confounded by the Wind PPA. A Wind PPA means that the wind power purchased by campus is not necessarily produced near the campus. In fact, it's produced in a different county than Champaign county. So improvements to the prediction by adding weather and solar angle will be difficult if we use "
```
[the remark this answers is not in the document]
Comment (samgdotson) [reply net_demand.ipynb:27]: PPA = power purchase agreement

File: twoinputs/netdemand-solarangle-hourly.ipynb (hunk 1 above)
Comment (samgdotson) [reply twoinputs/netdemand-solarangle-hourly.ipynb]: They did work, I accidentally re ran a simulation that I didn't want to run again. So I stopped the execution but all following cells appear "unexecuted."

Comment (samgdotson) [reply twoinputs/netdemand-solarangle-hourly.ipynb]: It's hard to tell what you mean by this in a Jupyter notebook. I couldn't find anything. Pep8 has different rules for indexing and math (but it also doesn't touch Jupyter notebooks anyway!).

File: net_demand.ipynb [hunk 3]
```diff
@@ -0,0 +1,1124 @@
...
...61YPuV/ncawd/lrieIF4EGtn6HSkf6+GxLYQQnB6I8+rN0hzxQXCYrlDKIm5dK5qHh0d7sZo3OXdrlXcd60JV2i9//OxwEinhtZs7qxJXBLFpmpimiaZpXFwpiam17tK+qzP4L75F/p3vxI0frMgN13VxHIexsbF1Va3u7m5cd32Vcy2VdvKTJ08yPj5OIpFoWxG8lrVzxJXfIZvNVm+ZTAbDMHbVVj0QKZ0bt7J3hXeqLI7vNXMuFoskEol9y2tOlgXxZn/n7SClbKognivP//fGGp+rjkQiFIvF6t9vLBng2koRpzyz+zfnFplZLvAzD/YwGK8tGKe6glxZKuxozjdXdDadH4a75lepFlaITdMkmUy2PHf7KBDQVIJ+tW7L9OKaCjFAyK/x/pM9vHRthVurxp6ss1XUE8TP0dhs8CfKx3p4bJvTA7Hql9NBqRBDydShViXGdlzPfdrD4xDy3StLSAnvarN26QrDHUE6I/4dt00bhoGmaetcl99acemPCKKVSp/jEPna13CSSYzHHm3GsveUXC7H0NDQhvnLYDBILBajWCxu+nN9fX0HLuPU5/MRDAbJZrPk83kCgQCDg4OcOHGC++67j6mpKTo6OnBdl0wmQyaT2fY8ddwPYd96Y620KYn4Bdo9G0eWZTEwMNCU320nxIMaliPJm7urmjuOg6ZpTTXUqhji9ccbbyWPx+MMDAxg2zb5fJ6xDh3TdrmdMfnuTIrnr6V5cirBg3WimKa6g5i2y40as8dbkTWdTR2moeQ+LQRkmpBFLKUkU7CrVefKfVLKfdtgOYwkQ35W6wjipayJqggSobseGh+Y7iXgU/jCAa8S19vi/D3gOSHE7wK/IaVc9y8khPABvwO8D3h365bocZhZG+qdPEiCOOir7uhWSBcs/v0zl7m1avB/ffwsShtWkTw8PLaPlJJnLy8y2RtpeMZvrxFC8OBwkm9dXKBgOduOsKs4TKfTaRRFYdFwWS5Inhi6+zzhb3wDdXmJ9E/9NByACulaisUiwWCQ3t7emo/39/fz5ptvbhDLpmni9/v3VcjthsnJSaSUBAKBDaZDwWCwmp9cLBbJ5/Pcvn2bTCZDIBBoqAIqhGAwolTniIUQpcile360UCgQjUb3dVMh4i/9/nnLIVxHyG1Fs+eHoVQhDviU6hxnIwghiEQiBINBVlZW6M0v4Lou37i
```
[the remark this answers is not in the document]
Comment (samgdotson) [reply net_demand.ipynb:1016]: Just testing different methods.

File: twoinputs/netdemand-solarangle-hourly.ipynb (hunk 1 above)
Comment (samgdotson) [reply twoinputs/netdemand-solarangle-hourly.ipynb]: Not intentional, but also not a problem. Just commenting an already commented line.

Comment (samgdotson) [reply twoinputs/netdemand-solarangle-hourly.ipynb]: All of the outputs should be there except for the cell I interrupted.

**Comment by samgdotson** [comment]
Thanks for the review, @LukeSeifert!

Yup, I did have a keyboard interrupt when I accidentally re ran a simulation. These notebooks take a long time to run, so I'm not going to completely re run them at this time. Eventually, I will get rid of the notebooks in favor of a script and snakemake.

**Comment by robfairh** [comment]
LGTM

**Pull request merged**